demo/app/lib/views/issuance_preview.dart

```diff
@@ -0,0 +1,256 @@
+                      ),
+                      PrimaryButton(
+                          onPressed: () async {
+                            if (widget.authorizeResultPinRequired == true) {
+                              navigateToOTPScreen(context);
+                            } else if (widget.uri != null) {
+                              navigateToAuthFlow(context, widget.uri!);
+                            } else {
+                              navigateToWithoutPinFlow(context);
+                            }
+                          },
+                          width: double.infinity,
+                          child: const Text('Add to Wallet', style: TextStyle(fontSize: 16, color: Colors.white))),
+                      const Padding(
+                        padding: EdgeInsets.fromLTRB(24, 0, 24, 8),
+                      ),
+                      PrimaryButton(
+                        onPressed: () {
+                          _navigateToDashboard();
+                        },
+                        width: double.infinity,
+                        gradient: const LinearGradient(
+                            begin: Alignment.topCenter,
+                            end: Alignment.bottomCenter,
+                            colors: [Color(0xffFFFFFF), Color(0xffFFFFFF)]),
+                        child: const Text('Cancel', style: TextStyle(fontSize: 16, color: Color(0xff6C6D7C))),
+                      ),
+                    ],
+                  ),
+                ), //last one
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void navigateToOTPScreen(BuildContext context) async {
+    Navigator.push(context, MaterialPageRoute(builder: (context) => const OTP()));
+  }
+
+  _navigateToDashboard() async {
+    Navigator.push(context, MaterialPageRoute(builder: (context) => const Dashboard()));
+  }
+
+  void navigateToAuthFlow(BuildContext context, Uri uri) async {
+    Navigator.of(context)
+        .push(MaterialPageRoute(builder: (context) => HandleRedirectUri(uri, "issuer-initiated-flow", "")));
+  }
+
+  void navigateToWithoutPinFlow(BuildContext context) async {
+    var credentialData = await fetchPreviewScreenDetails();
+
+    Navigator.push(context, MaterialPageRoute(builder: (context) => CredentialPreview(credentialData: credentialData)));
+  }
+
+  Future<CredentialData> fetchPreviewScreenDetails() async {
+    final Future<SharedPreferences> prefs = SharedPreferences.getInstance();
+    var WalletSDKPlugin = WalletSDK();
+    final StorageService storageService = StorageService();
+    final SharedPreferences pref = await prefs;
+
+    var didType = pref.getString('didType');
+    var keyType = pref.getString('keyType');
+    // choosing default if no selection is made
+    didType = didType ?? "ion";
+    keyType = keyType ?? "ED25519";
+
+    var didResolution = await WalletSDKPlugin.createDID(didType, keyType);
```

Comment:
Please refactor WalletSDKPlugin.createDID so it returns some structure as we did for other methods. To get rid of that encode decode part all over the code.

Reply:
Created this issue #534